### ISSUE TITLE
EditForm: change API to allow form subission from code

### ISSUE THREAD

**Issue by luisabreu** [dotnet/aspnetcore#48613]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Currentyle, it seems there's no way to submit an EditForm programatically from C# code because the HandleSubmitAsync method is private. 

### Describe the solution you'd like

Adding a public method that submits the form would allows us to have forms without submit buttons (ex.: scenario where the user must pick items from lists and where picking an item should submit a form) and would allow us to reuse the context+validation logic currently implemented by the EditForm.

**Comment by SteveSandersonMS** [comment]
To clarify, did you try triggering your own submit handler directly? That is, whatever method is wired up to the form - could you just call it manually? If your goal is to run validation, you can also run `yourEditForm.EditContext.Validate()` directly.

I'm asking to clarify whether there is some particular thing that can't be achieved this way, or whether it's just that you're looking for a more convenient built-in mechanism.

**Comment by luisabreu** [comment]
Yes, we're doing as you suggested: calling the validate and then calling the handler but it kind of breaks the flow, no? why can't we have a public submit method on the EditForm?

**Comment by SteveSandersonMS** [comment]
Sure, that's a totally reasonable request. I was only asking to clarify if an alternative existed and to make sure I wasn't missing something about what exactly you would expect the submit method to do.

**Comment by SteveSandersonMS** [comment]
I've moved this to backlog because we already have a full work schedule for .NET 8. However if you are keen enough to implement this and do a PR (including E2E tests) we would certainly consider taking it.

**Comment by SteveSandersonMS** [comment]
Just to add: this should probably work by making a JS interop call to instruct the browser to submit the form (instead of just invoking the .NET handler). That way any other expected client-side behaviors, such as triggering HTML's built-in validation prompts, should also occur.
